module_generator/generate_main.py

```diff
@@ -0,0 +1,103 @@
+    viam_module = __get_resource_class(resource_type, resource_name)
+    method_stubs = __generate_method_stubs(viam_module)
+
+    viam_class = "".join(word.capitalize() for word in resource_name.split("_"))
+
+    file_path = os.path.join(os.getcwd(), "src/__main__.py")
+    with open(file_path, 'w') as file:
+        file.write(f'''import asyncio
+from typing import ClassVar, Mapping, Optional, Self, Sequence, Union
+from viam.module.module import Module
+from viam.proto.app.robot import ComponentConfig
+from viam.proto.common import ResourceName
+from viam.resource.base import ResourceBase
+from viam.resource.easy_resource import EasyResource
+from viam.resource.types import Model, ModelFamily
+from viam.{resource_type}s.{resource_name} import {viam_class}
+
+
+class {model_name.capitalize()}({viam_class}, EasyResource):
+    """This is your custom {viam_class} {resource_type}.
+
+    All the methods listed are optional.
+    You can implement the methods you want and remove the methods you don't need.
+    """
+    MODEL: ClassVar[Model] = Model(ModelFamily("{namespace}", "{module_name}"), "{model_name}")
+
+    @classmethod
+    def new(cls, config: ComponentConfig, dependencies: Mapping[ResourceName, ResourceBase]) -> Self:
+        """This method creates a new instance of this vision service.
+        The default implementation sets the name from the `config` parameter and then calls `reconfigure`.
+
+        Args:
+            config (ComponentConfig): The configuration for this resource
+            dependencies (Mapping[ResourceName, ResourceBase]): The dependencies (both implicit and explicit)
+
+        Returns:
+            Self: The resource
+        """
+        return super().new(config, dependencies)
+
+    @classmethod
+    def validate_config(cls, config: ComponentConfig) -> Sequence[str]:
+        """This method allows you to validate the configuration object received from the machine,
+        as well as to return any implicit dependencies based on that `config`.
+
+        Args:
+            config (ComponentConfig): The configuration for this resource
+
+        Returns:
+            Sequence[str]: A list of implicit dependencies
+        """
+        raise NotImplementedError
+
+    def reconfigure(self, config: ComponentConfig, dependencies: Mapping[ResourceName, ResourceBase]):
+        """This method allows you to dynamically update your service when it receives a new `config` object.
+
+        Args:
+            config (ComponentConfig): The new configuration
+            dependencies (Mapping[ResourceName, ResourceBase]): Any dependencies (both implicit and explicit)
+        """
+        return super().reconfigure(config, dependencies)
+
+    {method_stubs}
+
+if __name__ == "__main__":
+    """This function will run your module and make it available to the machine."""
+    asyncio.run(Module.run_from_registry())
```

Comment:
When does `Registry.register_resource_creator` get run in order for `Module.run_from_registry` to pull this model from the registry?

Reply:
Right now I guess it doesn't. I haven't actually tested the module itself. Thanks for catching that!